tools/orca/src/generate.rs

```diff
@@ -156,6 +126,45 @@ pub async fn populate(_client: &KanidmOrcaClient, profile: Profile) -> Result<St
         persons.push(p);
     }
 
+    // Now, assign persons to roles.
+    //
+    // We do this by iterating through our roles, and then assigning
+    // them a baseline of required accounts with some variation. This
+    // way in each test it's guaranteed that *at least* one person
+    // to each role always will exist and be operational.
+
+    for group in groups.iter_mut() {
+        // For now, our baseline is 50%. We can adjust this in future per
+        // role for example.
+        let baseline = persons.len() / 2;
+        let inverse = persons.len() - baseline;
+        // Randomly add extra from the inverse
+        let extra = Uniform::new(0, inverse);
+        let persons_to_choose = baseline + seeded_rng.sample(extra);
+
+        assert!(persons_to_choose <= persons.len());
+
+        debug!(?persons_to_choose);
+
+        let person_index = index::sample(&mut seeded_rng, persons.len(), persons_to_choose);
+
+        // Order doesn't matter, lets optimise for linear lookup.
+        let mut person_index = person_index.into_vec();
+        person_index.sort_unstable();
+
+        for p_idx in person_index {
+            let person = persons.get_mut(p_idx).unwrap();
+
+            // Add the person to the group.
+            group.members.insert(person.username.clone());
+
+            // Add the reverse links, this allows the person in the test
+            // to know their roles
+            person.member_of.insert(group.name.clone());
```

Comment:
Since `member_of` and `roles` are so tightly correlated (we also have the `requires_membership_to` method on `ActorRole`), wouldn't it make sense to couple them together in a single struct/enum to avoid possible missmatches? ie we only keep the `roles` field and if we want to know the groups we're in we just use `requires_memebership_to`.

Reply:
I actually don't even think we need the "memberof" on the person tbh. I think we should remove it since ActorRole is wht defines what can be done.

Reply:
Yeah that’s what I was also thinking!